src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderAssessmentRepositoryTest.java

```diff
@@ -113,10 +88,73 @@ void getAssessmentByBookingIdAndAssessmentSeq() {
 
     @Test
     void getAssessmentByBookingIdAndAssessmentSeq_ReturnsNothing() {
-        final var assessment = repository.findByBookingIdAndAssessmentSeq(-43L, 4);
+        final var assessment = repository.findByBookingIdAndAssessmentSeq(-43L, 11);
 
         assertThat(assessment).isEmpty();
     }
+
+    @Test
+    void getAssessmentaByCsraAssessmentAndOffenderNo() {
+        final var assessments = repository.findByCsraAssessmentAndByOffenderNo("A1183JE");
+
+        assertThat(assessments).usingRecursiveComparison()
+            .ignoringFields("assessmentDate", "assessmentComment", "nextReviewDate", "assessCommittee", "assessStatus", "assessmentCreateLocation", "assessmentItems", "assessmentType",
+                "calculatedClassification", "creationUser", "evaluationDate", "modifyUserId", "offenderBooking",
+                "overrideReason", "overridingClassification", "reviewCommittee", "reviewCommitteeComment",
+                "reviewedClassification", "createDatetime", "createUserId")
+            .isEqualTo(List.of(
+                OffenderAssessment.builder()
+                    .bookingId(-43L)
+                    .assessmentSeq(1)
+                    .build(),
+                OffenderAssessment.builder()
+                    .bookingId(-43L)
+                    .assessmentSeq(2)
+                    .build(),
+                OffenderAssessment.builder()
+                    .bookingId(-43L)
+                    .assessmentSeq(3)
+                    .build()
+            ));
+
+        assertCsraAssessment_Booking43_AssessmentSeq2(assessments.get(1));
+    }
+
+    @Test
+    void getAssessmentaByCsraAssessmentAndOffenderNo_ReturnsNothing() {
+        final var assessments = repository.findByCsraAssessmentAndByOffenderNo("A1183JC");
+
+        assertThat(assessments).isEmpty();
+    }
+
+    private void assertCsraAssessment_Booking43_AssessmentSeq2(final OffenderAssessment assessment) {
+        assertThat(assessment.getBookingId()).isEqualTo(-43L);
```

Comment:
Might be worth a look at assertThat(assessment).extracting as it'll reduce the noise a bit.